ProjFS.Mac/PrjFSKext/KauthHandler.cpp

```diff
@@ -756,7 +844,7 @@ KEXT_STATIC bool ShouldHandleVnodeOpEvent(
         }
     }
 
-    *pid = GetPid(context);
+    *pid = vfs_context_pid(context);
```

Comment:
I think this is just cleanup.  Can you confirm?

Reply:
Yes, I noticed we were pointlessly replicating the kernel’s built-in `vfs_context_pid()` with our `GetPid` helper.

Reply:
Let's definitely take the change here, but if we can make separate small PRs for stuff like this it just makes reviewing easier.  And if we're slow at reviewing super small PRs with cleanup please do yell at us / me :-).

Reply:
As long as we don't save a `vfs_context_t` for use outside the kauth listener callback, and we likewise don't use any previously-saved VFS contexts in the callback, it doesn't really matter either way in practice. `proc_self` is implicit state of the execution context, while the `vfs_context_t` is an explicit reference to it, unless it's `nullptr`, in which case most functions fall back to `proc_self` etc.

One way in which the VFS context may matter is when performing I/O (including metadata I/O) on vnodes, because the context carries the permissions of the process it describes. So if the userspace process is for whatever reason not permitted to access a file in the way the kext tries to do, the kext will also be denied permission if it uses the standard context. I assume we haven't run into any edge cases where this matters yet, but in theory, we should look out for this, especially when populating the vnode cache - e.g. if we were denied permission to read the xattr of a directory to check if it's a root, we shouldn't then assume that it's not a root and save that state. The other option is to perform I/O with the kernel's root VFS context, which shouldn't have the permissions problem, but then you have to be careful about opening up information leakage security bugs.